src/commands/index.rs

```diff
@@ -147,16 +148,27 @@ where
         .and_then(|index| {
             let results: Result<Vec<_>, RedisError> = index
                 .search(&query)?
-                .map(|key| {
+                .filter_map(|key| {
                     let key = ctx.open_key_writable(&key);
-                    let value = match key.get_value::<RedisJSON>(&REDIS_JSON_TYPE)? {
-                        Some(doc) => doc.to_string(&path, Format::JSON)?.into(),
-                        None => RedisValue::None,
-                    };
-                    Ok(value)
+                    key.get_value::<RedisJSON>(&REDIS_JSON_TYPE)
+                        .and_then(|value| {
```

Comment:
I suggest extracting the contents of the `.and_then(|value| {` closure into a separate function for readability.
It could be called `path_to_json` or such.

It might be even better to move this code to a method of `RedisJSON` which would serialize a specific path.

This way the current code would become something like:

```
    map.get(&index_name)
        .ok_or("ERR no such index".into())
        .map(|schema| &schema.index)
        .and_then(|index| {
            let results: Result<Vec<_>, RedisError> = index
                .search(&query)?
                .filter_map(|key| {
                    let key = ctx.open_key_writable(&key);
                    key.get_value::<RedisJSON>(&REDIS_JSON_TYPE)
                        .and_then(value.serialize_path(&path))
                        .transpose()
                })
                .collect();
            Ok(results?.into())
        })
```

Reply:
@gavrie done